### ISSUE TITLE
false error warning when setting recipes folder

### ISSUE THREAD

**Issue by shining-cat** [nextcloud/cookbook#542]
**Description**
When I specify a storage folder for cookbook's data, I am confronted with a **false-positive** error popup

**Reproduction**
1. open cookbook settings
2. choose a folder (shared/not shared, with leading underscore or not, at the root level or deeper)
3. validate
4. popup opens claiming: "Could not set recipe folder to xxx"
5. There is no message from cookbook in the system logs
6. Reload the page, **the folder has actually been selected**, and test-adding recipe does create content in the file folder

**Expected behavior**
since the folder has indeed been selected: show a confirmation popup

**Screenshots**
![Screenshot 2021-01-19 111907](https://user-images.githubusercontent.com/19568399/105020886-2c299580-5a48-11eb-9e55-ee0c013e84fa.jpg)

**Browser**
Chrome Version 80.0.3987.149 (Official Build) (32-bit) (portableApp)

**Versions**
Nextcloud server version: 20.0.5
Cookbook version: 0.7.10

**Comment by christianlupus** [comment]
The problem seems to be the call to `loadAll()` in the [AppNavi.vue file](https://github.com/nextcloud/cookbook/blob/7f84cd3c7f0b76d057a95586d3360fb3d18c7135/src/components/AppNavi.vue#L301).

**Comment by seyfeb** [comment]
@christianlupus Do you know what the function call is supposed to achieve? ^^

**Comment by christianlupus** [comment]
I assume reloading the recipes after a reindex or would that not make any sense?

**Comment by seyfeb** [comment]
Okay, so reindexing happens in the backend. We would expect that when setting the new directory

1. the browser requests reindexing the database for the new folder
2. the server does the reindexing, then returns once it is finished
3. the browser redirects to the new recipe index

?

**Comment by christianlupus** [comment]
I would say that the change of the folder should trigger the reindex automatically. Then the Frontend needs to wait for the termination of that process (returning ok or so) and then a fresh copy of recipes should be fetched. That might be better off in the AppMain vue file as the first download is done there as well. It might be just a single event...

**Comment by seyfeb** [comment]
We could add a Vuex action to handle the frontend state for this

**Comment by christianlupus** [comment]
A quick check revealed that the reindexing [is already done when the folder is changed](https://github.com/nextcloud/cookbook/blob/master/lib/Controller/ConfigController.php#L69).

All it takes is:
- Replacing [this line](https://github.com/nextcloud/cookbook/blob/master/src/components/AppNavi.vue#L301) with something like `$this.$emit('reloadRecipes')`
- Moving the [fetching routines](https://github.com/nextcloud/cookbook/blob/master/src/components/AppIndex.vue#L25-L42) to `AddMain.vue`
- In `AppMain.vue` to implement a handler for that event to refetch the set of recipes

Alternatively, the refetching could be done in Vuex mutations that are called both from AppIndex as well as AppNavi.
